ARCHITECTURE.md

```diff
@@ -1,68 +1,3 @@
 # Architecture
 
-This document describes the high-level architecture of PostgREST.
```

Comment:
I thought the codemap was not too valuable (since I never got any feedback about it) and too easy to get out of sync. So I deleted it all.

But yeah maybe that was too stern.. I'll bring some text back and update.